External/FEXCore/Source/Utils/AllocatorOverride.cpp

```diff
@@ -0,0 +1,139 @@
+#include <FEXCore/Utils/Allocator.h>
+#include <FEXCore/Utils/CompilerDefs.h>
+#include <FEXCore/Utils/LogManager.h>
+
+#include <fmt/format.h>
+
+#include <cstddef>
+#include <cstdint>
+#include <fstream>
+#include <sstream>
+#include <unistd.h>
+
+extern "C" {
+#ifdef GLIBC_ALLOCATOR_FAULT
+  // glibc allocator faulting. If anything uses a glibc allocator symbol then cause a fault.
+  // Ensures that FEX only ever allocates memory through its allocator routines.
+  // If thunks are enabled then this should crash immediately. Thunks should still go through glibc allocator.
+  // These need to be symbol definitions /only/, don't let their visibility to the rest of the source happen.
+#define GLIBC_ALIAS_FUNCTION(func) __attribute__((alias(#func), visibility("default")))
+  extern void *__libc_calloc(size_t, size_t);
+  void *calloc(size_t, size_t) GLIBC_ALIAS_FUNCTION(fault_calloc);
+
+  extern void __libc_free(void*);
+  void free(void*) GLIBC_ALIAS_FUNCTION(fault_free);
+
+  extern void *__libc_malloc(size_t);
+  void *malloc(size_t) GLIBC_ALIAS_FUNCTION(fault_malloc);
+
+  extern void *__libc_memalign(size_t, size_t);
+  void *memalign(size_t, size_t) GLIBC_ALIAS_FUNCTION(fault_memalign);
+
+  extern void *__libc_realloc(void*, size_t);
+  void *realloc(void*, size_t) GLIBC_ALIAS_FUNCTION(fault_realloc);
+
+  extern void *__libc_valloc(size_t);
+  void *valloc(size_t) GLIBC_ALIAS_FUNCTION(fault_valloc);
+
+  extern int __posix_memalign(void **, size_t, size_t);
+  int posix_memalign(void **, size_t, size_t) GLIBC_ALIAS_FUNCTION(fault_posix_memalign);
+
+  extern size_t __malloc_usable_size(void*);
+  size_t malloc_usable_size(void*) GLIBC_ALIAS_FUNCTION(fault_malloc_usable_size);
+
+  // Reuse __libc_memalign
+  void *aligned_alloc(size_t, size_t) GLIBC_ALIAS_FUNCTION(fault_aligned_alloc);
+}
+
+namespace FEXCore::Allocator {
```

Comment:
The rest of this file needs *some* documentation.

Reply:
Added more comments.